### ISSUE TITLE
EF Core migration to MariaDB database fail

### ISSUE THREAD

**Issue by Ni-gashi** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#734]
### Steps to reproduce

1. Create MariaDB on Azure - [Azure MariaDB](https://docs.microsoft.com/en-us/azure/mariadb/tutorial-design-database-using-portal)
2. Create Console App (.NET Core) via Visual Studio 2017
3. Donwload NuGet package - Pomelo.EntityFrameworkCore.MySql
4. Create simple code for testing (DB context class and Program class is attached)

[Program.txt](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/files/2772864/Program.txt)
[TestClass.txt](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/files/2772866/TestClass.txt)

5. Create Migration via Package Manager Console in Visual Studio 2017 (Add-Migration Inititial)
6. Update database on Azure via Package Manager Console in Visual Studio 2017 (Update-Database)

### The issue
The migration was unsuccessful and failed.

```
Exception message: Failed to read the result set.
Stack trace:
MySql.Data.MySqlClient.MySqlException (0x80004005): Failed to read the result set. ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at MySqlConnector.Protocol.Serialization.ByteArrayReader.ReadLengthEncodedInteger() in C:\projects\mysqlconnector\src\MySqlConnector\Protocol\Serialization\ByteArrayReader.cs:line 124
   at MySqlConnector.Protocol.Serialization.ByteArrayReader.ReadLengthEncodedByteString() in C:\projects\mysqlconnector\src\MySqlConnector\Protocol\Serialization\ByteArrayReader.cs:line 155
   at MySqlConnector.Protocol.Payloads.OkPayload.Create(ReadOnlySpan`1 span, Boolean deprecateEof) in C:\projects\mysqlconnector\src\MySqlConnector\Protocol\Payloads\OkPayload.cs:line 47
   at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in C:\projects\mysqlconnector\src\MySqlConnector\Core\ResultSet.cs:line 47
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet(ResultSet resultSet) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 80
   at MySql.Data.MySqlClient.MySqlDataReader.ReadFirstResultSetAsync(IOBehavior ioBehavior) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 302
   at MySql.Data.MySqlClient.MySqlDataReader.CreateAsync(MySqlCommand command, CommandBehavior behavior, ResultSetProtocol resultSetProtocol, IOBehavior ioBehavior) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 287
   at MySqlConnector.Core.TextCommandExecutor.ExecuteReaderAsync(String commandText, MySqlParameterCollection parameterCollection, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\Core\TextCommandExecutor.cs:line 37
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlCommand.cs:line 261
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery() in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlCommand.cs:line 62
   at Pomelo.EntityFrameworkCore.MySql.Storage.Internal.MySqlDatabaseCreator.<>c__DisplayClass18_0.<Exists>b__0(DateTime giveUp)
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.<>c__DisplayClass12_0`2.<Execute>b__0(DbContext c, TState s)
   at Pomelo.EntityFrameworkCore.MySql.Storage.Internal.MySqlExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ExecutionStrategyExtensions.Execute[TState,TResult](IExecutionStrategy strategy, Func`2 operation, Func`2 verifySucceeded, TState state)
   at Microsoft.EntityFrameworkCore.Migrations.HistoryRepository.Exists()
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.UpdateDatabase(String targetMigration, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
```

### Further technical details

MariaDB version: 10.2.17
Pomelo.EntityFrameworkCore.MySql version: 2.1.4
Microsoft.NETCore.App version: 2.2.0

Other details about my project:
When I tried it on a local database (same version), everything worked. 
MariaDB was chosen because of Azure prices and I chose this NuGet package because it was recommended on official Microsoft websites.
When I tried to replace this NuGet package by MySql.Data.EntityFrameworkCore version 8.0.13, migration would fall on the error that there was no EFMigration table. When I created it manualy everything was fine.

**Comment by mguinness** [comment]
This could in the underlying library, see https://github.com/mysql-net/MySqlConnector/issues/498 for more information.  Try creating a simple console application using MySqlConnector to connect to Azure MariaDB.

**Comment by baltie** [comment]
I had this exact same problem connecting to an Azure MariaDB instance. Adding MySqlConnector as an explicit nuget reference, thereby upgrading it from 0.47.1 to 0.49.3, solved the problem for me.

**Comment by bgrainger** [comment]
The crash appears to be the problem raised in https://github.com/mysql-net/MySqlConnector/pull/603. This is fixed in MySqlConnector 0.49.2 and later.
